Fix `MenuBar` and `MenuButton` hover position scaling properly with the scale factor multiplier

OverloadedOrama opened the pull request:
Fixes #86240. The solution was basically to just remove `mouse_pos_adjusted` and the code related to it, and simply get the position from `get_local_mouse_position()`, and it now works properly. I don't think this breaks anything, but feel free to correct me if I'm wrong. From what I see, the code is there since the beginning of `MenuBar` (#63950) and it seems to be copy-pasted from the `MenuButton` code, introduced in #50619.

~~`MenuButton` seems to be a bit more difficult to fix. Simply using `get_local_mouse_position()` is not working properly, but getting the local mouse position from the parent is working properly. Problem is, this function is only implemented in `CanvasItem` nodes. If the `MenuButtons` have a 3D node as a parent, this solution is not going to work. Which is why, I am assuming, the `mouse_pos_adjusted` code is in `MenuButton` in the first place. Any ideas on how to handle this would be appreciated!~~
EDIT: `MenuButton` has also been fixed, thanks to #78472.

If this PR gets accepted, I would greatly appreciate it if also got cherry-picked to 4.2.x.

OverloadedOrama committed:
Fix `MenuBar` hover position scaling properly with the scale factor multiplier

Sauermann commented:
During my investigation into how to fix this, i came to a nearly identical conclusion. Please have a look at #78472.

OverloadedOrama commented:
Oh nice, I didn't know of this PR. It seems that our code is identical when it comes to MenuBar, but yours addresses even more issues. I guess this PR should be closed then, and hopefully yours will get merged soon and ideally cherry-picked for 4.2.2. Thanks for bringing it into my attention!

Sauermann commented:
Lets's leave your PR open, since it is easier to review.

OverloadedOrama commented:
> > It seems that our code is identical when it comes to MenuBar
> 
> Our approaches differ: in your code `set_process_internal(true);` gets called only when the MenuBar is within a `Window` while in my code it gets called additinally when it is within a `SubViewport`.

Right, I didn't notice that. I only changed the part related to the scale factor multiplier issue, I left the rest of the code as-is. But it sounds to me that your approach is even better, as it makes MenuBar hovering work while inside `SubViewport`s too. Both approaches fix #86240 though.

OverloadedOrama commented:
I updated the PR to also fix `MenuButton`'s scaling issue, and I updated the `MenuBar` code changes to be the same as #78472, so our PRs should be identical now for MenuButton and MenuBar, in case the maintainers decide they want to merge this one first.

Sauermann reviewed (approved):
Regarding MenuBar: I came to the same conclusion of how to fix the problem.
Regarding MenuButton: Additionally someone else needs to verify the changes, since they are identical to my suggested approach in #78472.

akien-mga reviewed (approved):
Looks good to me.

akien-mga commented:
Thanks!

akien-mga commented:
Cherry-picked for 4.2.2.
